locksmith/src/operations/receiptOperations.ts

```diff
@@ -1,29 +1,74 @@
 import { SubgraphService } from '@unlock-protocol/unlock-js'
 import { Receipt, ReceiptBase } from '../models'
+import { getMetadataByTokenId } from './userMetadataOperations'
 
 interface ReceiptDetailsProps {
   lockAddress: string
   network: number
   hash: string
+  tokenId?: string
 }
 
-export const getReceiptDetails = async ({
+/** Get purchaser details from metadata if email is collected otherwise return from receipt */
+const getPurchaserDetails = async ({
   lockAddress,
   network,
   hash,
-}: ReceiptDetailsProps): Promise<{
-  supplier: ReceiptBase | null
-  purchaser: Receipt | null
-  receipt: any
-}> => {
-  const purchaser = await Receipt.findOne({
+  tokenId,
+}: ReceiptDetailsProps): Promise<Partial<Receipt> | null> => {
+  const receiptPurchaser = await Receipt.findOne({
     where: {
       network,
       lockAddress,
       hash,
     },
   })
 
+  let metadata: any
+  if (tokenId) {
+    metadata = await getMetadataByTokenId({
+      tokenId,
+      network,
+      lockAddress,
+      includeProtected: true,
+    })
+  }
+
+  const data = metadata?.userMetadata?.protected ?? {}
+
+  // return metadata if email is collected
+  if (data?.email) {
```

Comment:
@julien51 need to check email email? 
checkout can collect `metadata` but not the email, we need to return the following when `receipt data` are not present event if we don't have the email?